x/twap/logic_test.go

```diff
@@ -521,3 +522,422 @@ func (s *TestSuite) TestPruneRecords() {
 
 	s.validateExpectedRecords(expectedKeptRecords)
 }
+
+// TestUpdateRecords tests that the records are updated correctly.
+// It tests the following:
+// - two-asset pools
+// - multi-asset pools
+// - with spot price errors
+// - without spot price errors
+// - that new records are created
+// - older historical records are not updated
+// - spot price errot times are either propagated from
+// older records or set to current block time in case error occurred.
+func (s *TestSuite) TestUpdateRecords() {
```

Comment:
I think we have duplicated boundaries of testing with `TestUpdateTwap` with this test case that could be merged into one, wdyt?

Reply:
`TestUpdateTwap` tests `updateRecord` method while this is testing `updateRecords` (plural).

I've referenced that test while implementing it, and, unfortunately, I could not find any logic suitable for reuse or de-duplication.

Reply:
Ah didn't notice the difference in test being plural -and singular. IMO we should change `TestUpdateTwap` -> `TestUpdateRecords` so that we can be consistent with the testing names